internal/worker/removal/worker.go

```diff
@@ -76,3 +169,32 @@ func (w *removalWorker) Kill() {
 func (w *removalWorker) Wait() error {
 	return w.catacomb.Wait()
 }
+
+// jobWorker exists only to expose enough of a tomb to satisfy [worker.Worker]
+type jobWorker struct {
+	tomb tomb.Tomb
+}
+
+// newJobWorker returns a closure suitable for passing to
+// a runner's StartWorker method.
+// It uses the input service to run the input removal job.
+func newJobWorker(svc RemovalService, job removal.Job) func() (worker.Worker, error) {
+	return func() (worker.Worker, error) {
+		w := jobWorker{}
+		w.tomb.Go(func() error {
+			return svc.ExecuteJob(w.tomb.Context(context.Background()), job)
+		})
+		return &w, nil
+	}
```

Comment:
It would be nice for the `remove.Job` to be stored in the `jobWorker`, such that we could implement a `Report` method.